src/database/ADODB_base.php

```diff
@@ -184,10 +184,10 @@ public function fieldClean(&$str)
         /**
          * Cleans (escapes) a string
          *
-         * @param $str The string to clean, by reference
-         * @return The cleaned string
+         * @param  string $str The string to clean, by reference
+         * @return string The cleaned string
          */
-        public function clean(&$str)
+        public function clean(string &$str)
```

Comment:
@richp10 it seems this type hinting is causing trouble.

See, sometimes `clean` method is used with null parameters, so it's crashing for example when you create a primary key and leave the index name field empty (which is allowed according to Postgres Docs section [sql-altertable.html](https://www.postgresql.org/docs/9.6/static/sql-altertable.html))

I believe this is the only scalar typehinting you added, so I removed it in the last tag